### ISSUE TITLE
Field.sortAsc() should not hard wire NULLS FIRST behaviour

### ISSUE THREAD

**Issue by sonkat** [jOOQ/jOOQ#13085]
### Expected behavior
When we use sortAsc, we expect it to generate the SQL as documented: 
https://www.jooq.org/javadoc/3.15.4/org.jooq/org/jooq/Field.html#sortAsc(T...)

```sql
CASE [this] WHEN [sortList.get(0)] THEN 0
             WHEN [sortList.get(1)] THEN 1
             ...
             WHEN [sortList.get(n)] THEN n
             ELSE null
 END ASC
```

### Actual behavior

```sql
CASE WHEN [this] is not distinct from [sortList.get(0)] THEN 1
          ...
         ELSE 0
 END ASC
```

### Steps to reproduce the problem
Statement in Java:

```java
dsl().select(ID)
.from(TEST)
.orderBy(TEST.INSTANCE_ID.sortAsc(sourceInstanceId), TEST.SORT_ORDER)
.fetchInto(SourceInstanceId.class)
```

Correct behavior in 3.11 - generated code:

```sql
select "test"."id"
from "test"
order by
case "test"."instance_id"
when 'XXX_ID' then 0 end asc,
"test"."sort_order"
```

Changed behavior in 3.15 - generated code:

```sql
select "test"."id"
from "test"
order by
case
when "test"."instance_id" is not distinct from 'XXX_ID' then 1
else 0
end asc,
"test"."sort_order"
``` 

## Generated code configuration (3.11):

```
generate([:]) {
                            records true
                            pojosEqualsAndHashCode mode.generatePojos
                            daos mode.generateDaos
                            immutablePojos false
                            jpaAnnotations false
                            validationAnnotations false
                            interfaces false
                            fluentSetters true
                            javaTimeTypes true
                        }
```

## Generated code configuration (3.15):

```
generate([:]) {
                            records true
                            pojosEqualsAndHashCode mode.generatePojos
                            daos mode.generateDaos
                            immutablePojos false
                            jpaAnnotations false
                            validationAnnotations false
                            interfaces false
                            fluentSetters true
                            javaTimeTypes true
                            printMarginForBlockComment 0
                            comments false
                            javadoc false
                            varargSetters true
                            fullyQualifiedTypes ".*_DATA_API"
                            generatedAnnotation true
                            generatedAnnotationType "JAVAX_ANNOTATION_GENERATED" 
                            generatedSerialVersionUID "HASH"
                        }
```

### Versions of correct behavior

- jOOQ: 3.11.11
- Java: 13
- Database (include vendor): PostgreSQL 11
- OS: Windows, Linux, Mac
- JDBC Driver (include name if inofficial driver): 42.2.5

### Versions of new behavior

- jOOQ: 3.15.4
- Java: 13
- Database (include vendor): PostgreSQL 13
- OS: Windows, Linux, Mac
- JDBC Driver (include name if inofficial driver): 42.3.1

**Comment by lukaseder** [comment]
Thanks for your report. The change was probably introduced in jOOQ 3.15 by https://github.com/jOOQ/jOOQ/issues/11009.

While the Javadoc helps document what happens, there's no need for an implementation to be bound by this as a "contract." Why is the old behaviour important to you? I.e. why do you consider this to be a bug? (Apart from the Javadoc being outdated)?

**Comment by sonkat** [comment]
@lukaseder - I consider this a bug, as it doesn't sort the entries in the same order. 

Say we have a table test with the following entries:
'YYY_ID', 'Test_lokal1'
'YYY_ID', 'Test_lokal2'
'XXX_ID', 'Test_nichtlokal'

case "test"."instance_id" when 'XXX_ID' then 0 end asc,
--> Result of table:
'XXX_ID', 'Test_nichtlokal'
'YYY_ID', 'Test_lokal1'
'YYY_ID', 'Test_lokal2'

case when "test"."instance_id" is not distinct from 'XXX_ID' then 1 else 0
--> Result of table:
'YYY_ID', 'Test_lokal1'
'YYY_ID', 'Test_lokal2'
'XXX_ID', 'Test_nichtlokal'

**Comment by lukaseder** [comment]
Oh I see now, thanks for the clarification. Yes, the refactoring shouldn't hard-wire the `NULLS FIRST` behaviour. Before, it was unspecified and defaulted to the dialect's default for `NULL` sorting. In addition to that, `nullsFirst()` and `nullsLast()` no longer has any effect. Definitely a bug.

**Comment by lukaseder** [comment]
I see, the `nullsFirst()` and `nullsLast()` integration tests were skipped because of an unrelated test problem, which is why this regression remained undiscovered.

A workaround is to use the `sort(Map)` overload, and pass a `Map` like this:

```java
Map.of(field1, inline(1), field2, inline(2), ...)
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.5 (https://github.com/jOOQ/jOOQ/issues/13086), and 3.15.9 (https://github.com/jOOQ/jOOQ/issues/13087).

Thanks again for your report!
